indico/modules/events/editing/client/js/management/ManageService.jsx

```diff
@@ -0,0 +1,228 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2020 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import checkServiceURL from 'indico-url:event_editing.api_check_service_url';
+import checkServiceStatusURL from 'indico-url:event_editing.api_service_status';
+import connectServiceURL from 'indico-url:event_editing.api_service_connect';
+import disconnectServiceURL from 'indico-url:event_editing.api_service_disconnect';
+
+import React, {useState} from 'react';
+import PropTypes from 'prop-types';
+import {FORM_ERROR} from 'final-form';
+import {Form as FinalForm} from 'react-final-form';
+import {Confirm, Modal, Button, Form, Loader, Message} from 'semantic-ui-react';
+import {Translate, Param} from 'indico/react/i18n';
+import {
+  FinalSubmitButton,
+  FinalInput,
+  handleSubmitError,
+  validators as v,
+} from 'indico/react/forms';
+import {indicoAxios, handleAxiosError} from 'indico/utils/axios';
+import {useIndicoAxios} from 'indico/react/hooks';
+import Section from './Section';
+
+export default function ManageService({eventId}) {
+  const [serviceURLInfo, setServiceURLInfo] = useState(false);
+  const [connectOpen, setConnectOpen] = useState(false);
+  const [disconnectOpen, setDisconnectOpen] = useState(false);
+  const {data, error, loading, reFetch} = useIndicoAxios({
+    url: checkServiceStatusURL({confId: eventId}),
+    trigger: eventId,
+    camelize: true,
+  });
+
+  const checkPending = loading || (data === null && error === null);
+  let label = Translate.string('Custom editing workflow');
+  let description;
+  let showConnect = false;
+  let showDisconnect = false;
+  let forceDisconnect = false;
+
+  if (checkPending) {
+    description = Translate.string('Checking custom editing workflow status');
+  } else if (error || data.error) {
```

Comment:
Do we need to double-check for the error? Isn't it already guaranteed to be true if `data.error` exists?

![image](https://user-images.githubusercontent.com/12183954/79861993-3d684780-83cd-11ea-9c62-23f8b685c469.png)

Reply:
`error` means the request to our own backend failed, `data.error` means that checking the service itself failed. You never have both at the same time.

In any case, we should never end up here with neither an error nor data (since we're in checkPending in that case)

Reply:
I didn't realize earlier that `data.error` was referring to the service. That makes sense.